mesonbuild/modules/gnome.py

```diff
@@ -307,17 +309,18 @@ def _print_gdbus_warning() -> None:
     @staticmethod
     def _find_tool(state: 'ModuleState', tool: str) -> 'ToolType':
         tool_map = {
-            'gio-querymodules': 'gio-2.0',
-            'glib-compile-schemas': 'gio-2.0',
-            'glib-compile-resources': 'gio-2.0',
-            'gdbus-codegen': 'gio-2.0',
-            'glib-genmarshal': 'glib-2.0',
-            'glib-mkenums': 'glib-2.0',
-            'g-ir-scanner': 'gobject-introspection-1.0',
-            'g-ir-compiler': 'gobject-introspection-1.0',
+            'gio-querymodules': ('gio-2.0', 'gio_querymodules'),
+            'glib-compile-schemas': ('gio-2.0', 'glib_compile_schemas'),
+            'glib-compile-resources': ('gio-2.0', 'glib_compile_resources'),
+            'gdbus-codegen': ('gio-2.0', 'gdbus_codegen'),
+            'glib-genmarshal': ('glib-2.0', 'glib_genmarshal'),
+            'glib-mkenums': ('glib-2.0', 'glib_mkenums'),
+            # TODO: Use gi_compile_repository once GLib exposes it
+            'gi-compile-repository': ('girepository-2.0', None),
```

Comment:
The code as-written does not allow looking up tools which are not in pkg-config, this adapts the code to allow finding tools via binary lookup only

Reply:
1. That's just `state.find_program()` instead of `state.find_tool()`...
2. Upstream MR has been merged (not released) so I guess we can simplify this again.